src/NumericalAlgorithms/Interpolation/Callbacks/ObserveSurfaceIntegrals.hpp

```diff
@@ -33,33 +38,81 @@ namespace db {
 template <typename TagsList>
 class DataBox;
 }  // namespace db
+namespace observers {
+namespace ThreadedActions {
+struct WriteReductionData;
+}  // namespace ThreadedActions
+template <class Metavariables>
+struct ObserverWriter;
+}  // namespace observers
 /// \endcond
 
 namespace intrp {
 namespace callbacks {
 
+namespace detail {
+template <typename T>
+struct reduction_data_type;
+
+template <typename... Ts>
+struct reduction_data_type<tmpl::list<Ts...>> {
+  // We use ReductionData because that is what is expected by the
+  // ObserverWriter.  We do a "reduction" involves only one core,
+  // so AssertEqual is used here as a no-op.
+  //
+  // The first argument is for Time, the others are for
+  // the list of scalars being integrated.
+  using type = Parallel::ReductionData<
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<typename Ts::type::type::value_type,
+                               funcl::AssertEqual<>>...>;
+};
+
+template <typename T>
+struct reduction_data_tag_type;
+
+template <typename... Ts>
+struct reduction_data_tag_type<tmpl::list<Ts...>> {
+  // The first argument is for Time, the others are for
+  // the list of scalars being integrated.
+  using type = tmpl::list<observers::Tags::ReductionData<
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
```

Comment:
No, this is the time that gets written to the file (not the internally stored timestep), which as I understand it will always be a double even with global timestepping (so I think, @wthrowe ?). This is a TimeId before it is written.

Reply:
I haven't figured out a great way to represent arbitrary ID types in HDF5 files other than writing them as a `double`. It might be possible but it's also not clear it'll be necessary unless we need to be able to use HDF5 files for check pointing type work.

Reply:
A double is correct here.